base_model_restrict_update/models/ir_model_access.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, models, tools
+from odoo.exceptions import AccessError
+from odoo.tools.translate import _
+
+
+class IrModelAccess(models.Model):
+    _inherit = "ir.model.access"
+
+    @api.model
+    @tools.ormcache_context(
+        "self._uid", "model", "mode", "raise_exception", keys=("lang",)
+    )
+    def check(self, model, mode="read", raise_exception=True):
+        if self.env.su:
+            return True
+        res = super().check(model, mode, raise_exception)
+        if mode != "read" and raise_exception:
+            if self._test_readonly(model) or self._test_restrict_update(model):
```

Comment:
@kittiu Could we exclude some models from this checking (e.g. a read-only user may want to create a new custom filter)? ref. https://gist.github.com/moylop260/43748e848dfc9916fd2f0fd51a69a12a#file-read_only_user-py-L11-L18

Thank you very much for the nice addition.

Reply:
Hi Yoshi, let's discuss together what should be excluded. I am actually ok with your proposal.
(this means bring back https://github.com/OCA/server-tools/compare/44005c10f2854150dd80755ed49852d8ff91bafb..130ca667f8f9c5c4b855e06d35fcf1ac07a99024)

At first I think readyonly is readonly. But it also make sense for, i.e., readonly auditor to have own filter.

@dsolanki-initos  @eLBati what is your opinion on this? If we can't tell what else, we can start with only ir.filters and add more excluded model later if needed.

Reply:
One way to go, I think, is to get a list of models through raw SQL based on create access = true and group is not set, which should be how @moylop260 came up with the list in the gist.

There are 161 such models in 14.0 EE as I look in runbot.
![image](https://user-images.githubusercontent.com/7766939/140255927-2a1063db-f963-494f-99dc-c7922eb71385.png)

Reply:
@yostashiro @kittiu thanks, it is clear the purpose of `_readonly_exclude_models` now.
I restored it and refactored to dinamically retrieve model names, see https://github.com/OCA/server-tools/pull/2188/commits/c69eba0fd0f4497c694fd7dbe16f56fff795acf7

Reply:
@eLBati I saw your excluded model.
```
        return self.sudo().search([
            ("group_id", "=", False),
            "|", ("perm_write", "=", True),
            "|", ("perm_create", "=", True), ("perm_unlink", "=", True)
        ]).mapped("model_id.model")
```
Not sure what is the rationale. If you find all ACL group_id = False with updatable flag, that will open a lot of model for update, isn't it? While, I think we have this module to restricted those models.
Please feel free to discuss, I never put this to real use and may not know exact problem.

Reply:
@kittiu I retrieve all the access rules not associated to a group where create, write or unlink is allowed.
That is, models where everyone should be able to create, write or delete.

On a v12, this is what it returns:
```
>>> env["ir.model.access"].search([
...     ("group_id", "=", False),
...     "|", ("perm_write", "=", True),
...     "|", ("perm_create", "=", True), ("perm_unlink", "=", True)
... ]).mapped("model_id.model")
['ir.ui.view.custom', 'res.users.log', 'report.layout', 'web_editor.converter.test', 'web_editor.converter.test.sub', 'decimal.precision.test', 'ir.attachment.add_url', 'mail.thread', 'publisher_warranty.contract']
```

Reply:
@eLBati I think the first intention is to not allow any edition to any model (as much as it could), i..e, most of the returns you show shouldn't be editable as well.
So I think to be compromised for everyone requirement, I think we start with a minimum list and make the list extensible on custom module.

I would start with,
- ir.filters (auditor should be able to create his/her own filter)

WDYT?

Reply:
I made https://github.com/OCA/server-tools/commit/c69eba0fd0f4497c694fd7dbe16f56fff795acf7 because I got into errors while a read-only user logs in and gets error because not able to create/write `res.users.log`.
So, the minimum set for me, AFAIK now, is `['res.users.log']`.
Thus, I made the `search` implementation instead of a fixed list, to let the system tell us which models to allow.

The other way I see is to restore your list
```
        [
            "res.users",
            "res.users.log",
            "mail.channel",
            "mail.channel.partner",
        ]
```

Other models like `ir.filters` should be considered in a second step I think; considering however that, with the `search` implementation, access could be granted modifying ACL

Reply:
IMO @eLBati 's approach is more generic and therefore is better. To me, what's globally updatable in the first place in the ACL should remain updatable even for read-only users. I think you can override `_readonly_exclude_models()` if there is a situation that goes against this idea.

@kittiu Can you think of a case where you may have some problem with above approach?

Reply:
Alright, I understand the point of @eLBati now. It is in a way configurable also. So I restored back using search()
Thanks both.